infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/workflows/AddNodeWorkflow.java

```diff
@@ -48,12 +47,16 @@
      *
      * @param request request to add a node
      */
-    public AddNodeWorkflow(AddNodeRequest request) {
+    public AddNodeWorkflow(AddNodeRequest request, StreamLog streamLog) {
         this.id = UUID.randomUUID();
         this.request = request;
         actions = ImmutableList.of(new BootstrapNode(),
                 new AddNodeToLayout(),
-                new RestoreRedundancy());
+                RestoreRedundancyMergeSegments.builder()
```

Comment:
For a sanity check, make sure that request.getEndpoint is equal to localEndpoint